tests/integration/pkg/helpers/api_rule.go

```diff
@@ -58,25 +58,26 @@ func ApplyApiRule(toExecute RetryableApiRule, onRetry RetryableApiRule, k8sClien
 				return err
 			}
 			if apiStatus.Status.APIRuleStatus.Code == "ERROR" {
-				log.Println("APIRule status not ok: " + apiStatus.Status.APIRuleStatus.Description)
-				return errors.New("APIRule status not ok: " + apiStatus.Status.APIRuleStatus.Description)
+				log.Printf("APIRule status is ERROR: %s", apiStatus.Status.APIRuleStatus.Description)
+				return fmt.Errorf("APIRule status is ERROR: %s", apiStatus.Status.APIRuleStatus.Description)
 			}
 			return nil
 		}, retryOpts...)
 	}
 	return nil
 }
 
-func ApplyApiRuleV2Alpha1(toExecute RetryableApiRule, onRetry RetryableApiRule, k8sClient dynamic.Interface, retryOpts []retry.Option, resources []unstructured.Unstructured) error {
+func ApplyApiRuleV2Alpha1RetryOnError(toExecute RetryableApiRule, onRetry RetryableApiRule, k8sClient dynamic.Interface, retryOpts []retry.Option, resources []unstructured.Unstructured) error {
```

Comment:
nit: What is the reasining of changing the name of a function? There is no point in stating that the function retries. If there is a godoc missing, please add it.